range_queries/mo_algorithm.cpp

```diff
@@ -0,0 +1,108 @@
+/**
+ *
+ * @file
+ * \brief A C++ Program to compute unique elements of ranges for different range
+ * queries
+ *
+ * \details
+ * The preprocessing part takes O(m Log m) time.
+ * Processing all queries takes O(n * √n) + O(m * √n) = O((m+n) * √n) time.
+ */
+#include <algorithm>
+#include <iostream>
+#include <cmath>
+#include <array>
+const int N = 1e6 + 5;
+std::array<int, N> a, bucket, cnt;
+// Variable to represent block size. This is made global so compare() of sort
+// can use it.
```

Comment:
Remember, no *global* variables. Instead, use them in a class or as function parameters.